src/contradictory_claims/data/process_claims.py

```diff
@@ -0,0 +1,162 @@
+    # I believe this should work, I wonder if it's not recommended for  memory reasons though in a v env like Travis...
+    nlp = spacy.load("en_core_sci_lg")
+    # Enable umls entity detection and abbreviation detection
+    linker = UmlsEntityLinker(resolve_abbreviations=True)
+    nlp.add_pipe(linker)
+    abbreviation_pipe = AbbreviationDetector(nlp)
+    nlp.add_pipe(abbreviation_pipe)
+
+    # Create a new vector to assign to the virus terms
+    new_vector = nlp("""Positive-sense single‐stranded ribonucleic acid virus, subgenus """
+                     """sarbecovirus of the genus Betacoronavirus. """
+                     """Also known as severe acute respiratory syndrome coronavirus 2, """
+                     """also known by 2019 novel coronavirus. It is """
+                     """contagious in humans and is the cause of the ongoing pandemic of """
+                     """coronavirus disease. Coronavirus disease 2019 is a zoonotic infectious """
+                     """disease.""").vector
+
+    # Add virus terms to the model vocabulary and assign to them the new vector created above
+    # vocab = Vocab()
+    virus_words = pd.read_csv(virus_lex_path, header=None)
+    for virus_word in virus_words[0]:
+        nlp.vocab.set_vector(virus_word, new_vector)
+
+    return nlp
+
+
+def tokenize_section_text(input_data: pd.DataFrame):
+    """
+    Tokenize section text to sentences.
+
+    :param input_data: pandas dataframe with publication text
+    :retunr: Dataframe with section text tokenized to sentences
+    """
+    # Empty dictonary to store the tokenized text
+    text_dict = {}
+    # Dictionary iterator
+    k = 0
+
+    # Loop through the sections and tokenize text to sentences
+    for i, text in enumerate(input_data.text):
+        for sent in sent_tokenize(text):
+            text_dict[k] = {'cord_uid': input_data.cord_uid[i],
+                            'section': input_data.section[i],
+                            'text': input_data.text[i],
+                            'drug_terms_used': input_data.drug_terms_used[i],
+                            'claims': sent}
+            k = k + 1
+
+    return pd.DataFrame.from_dict(text_dict, "index")
+
+
+def pair_similar_claims(claims_data: pd.DataFrame, nlp):
+    """
+    Pair similar claims.
+
+    :param claims_data: pandas dataframe with cord 19 claims
+    :param nlp: Scispacy nlp object
+    :return: Dataframe of paired claims
+    """
+    # Extract list of drug terms present across all claims
+    # Note: 'drug_terms_used' consists of drug terms present in the section in which the claim appears
+    drug_terms = []
+    for drugs in claims_data.drug_terms_used:
+        drug_terms = drug_terms + str(drugs).split(',')
+    drug_terms = list(set(drug_terms))
+    drug_terms.append('acei/arb')
+
+    # Filter to claims that contain drug terms
+    sentences_to_keep = [any(True for d in drug_terms if d in c) for c in claims_data.claims]
```

Comment:
Yes, this is quite a bottleneck. Let's discuss, I'm not familiar with hashing/set comparisons really so I'll look into that.

Reply:
If you're worried about missing on claims that don't contain the drug term explicitly, one thing to think about is the clarity in downstream classification/annotation/meta analysis. It can be confusing to evaluate when the claim doesn't have any drug term in it.